### DISCUSSION TITLE
getServerSideProps execution at build time

### DISCUSSION THREAD

**Discussion by singularitypress** [vercel/next.js#61675] [Help] [answered]
### Summary

Like the title says, I need some help understanding some differences in how the [documentation](https://nextjs.org/docs/pages/building-your-application/rendering/server-side-rendering) describes the behaviour of `getServerSideProps` and what I'm actually observing. Per the documentation, it specifically states (emphasis mine):
> `getServerSideProps ` is run on every request _instead_ of on build time.

What I'm observing though is that running the build is failing because a page using `getServerSideProps` is executing a function that references an environment variable that only exists in the main runtime container and not in the build container. Further, if I add some lots to the aforementioned function (that only runs in `getServerSideProps`, I see the logs execute when running my build. Is this a problem that only exists for projects with custom servers or is there something I'm missing (very possible tbh)?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I think it is often easy to get confused between, executing and parsing, bundling and such...  refer to:

- https://github.com/vercel/next.js/discussions/35534#discussioncomment-6084034
- https://github.com/vercel/next.js/discussions/35534#discussioncomment-2420828

So for example, top module code that is referenced by this page using getServerSideProps, might still fail, because when Next.js is building the page, it loads the the module, and its imports, to figure out what kind of work is required for it. 

Before I go on, do you feel like this is somewhat explaining your issue?
